backend/src/Book/Infrastructure/BookImporter.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Book\Infrastructure;
+
+use App\Book\Application\StoreBook;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+
+class BookImporter
+{
+    public function __construct(
+        private readonly JsonSchemaValidator $jsonSchemaValidator,
+        private readonly StoreBook $storeBook,
+        private readonly FileHandler $fileHandler
+    ) {
+    }
+
+    public function import(string $fileName): void
+    {
+        $filePath = realpath(\dirname(__DIR__)."/../../public/{$fileName}");
+        $file = fopen($filePath, 'r');
+        
+        // create a copy of the original file
+        if (!file_exists($filePath . ".copy")) {
+            copy($filePath, $filePath . ".copy");
+        }
+        
+        [$headers, $data, $dataLen] = $this->fileHandler->csvToArray($file, $filePath);
+        
+        fclose($file);
+    
+        for ($i = 0; $i < $dataLen; $i++) {
+            if ($this->validateRowData($data, $i)) {
```

Comment:
**nice-to-have:**  invece di avere `$this->validateRowData($data, $i)` sarebbe meglio avere `$this->validateRowData($data[$i])` in modo che il metodo `validateRowData` ha meno visibilità sulla struttura del dato e che faccia solo la validazione della riga che gli viene fornita

Reply:
done! 😊